### PR TITLE
Feature/nmbs 1998/add images support

### PR DESCRIPTION
**Pull request by QuentinVaucherLeanIX**

### CODE REVIEW

File: src/service/leanix-region.service.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { SecretClient } from '@azure/keyvault-secrets';
 import { throwErrorAndExit } from '../errors';
 import { getMtmToken } from '../client/mtm.client';
 
-export class LeanixRegionClient {
+export default class LeanixRegionService {
```
Comment (marvinrohde-leanix) [inline src/service/leanix-region.service.ts:6]: Did you encounter an issue with non-default classes apart that you need to wrap in curly braces when importing them?

File: src/service/integration.service.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      const integrationId = (await this.integrationClient.getIntegrations(regionId, token)).find((it) => it.name === integrationName)?.id;
+      if (integrationId === undefined) {
+        throw new Error(`Integration with name '${integrationName}' does not exist.`);
+      }
+
+      const assetsNames = fs.readdirSync(assetFolder);
+      for (const assetName of assetsNames) {
+        const assetPath = `${assetFolder}/${assetName}`;
+        await this.integrationClient.upsertAsset(regionId, integrationId, token, assetPath);
+        core.info(`Assets '${assetName}' posted successfully to region ${region}`);
```
Comment (marvinrohde-leanix) [inline src/service/integration.service.ts:37]: 🤓 
```suggestion
        core.info(`Asset '${assetName}' posted successfully to region ${region}`);
```

**Review by marvinrohde-leanix — COMMENTED** [review]
Looks very promising 🙂

File: src/client/integration.client.ts [hunk 3]
```diff
@@ -47,12 +50,47 @@ export class IntegrationClient {
...
+  async upsertAsset(regionId: string, integrationId: string, token: string, assetPath: string): Promise<IntegrationAssetResponseDto> {
+    const formData = new FormData();
+    formData.append('asset', fs.createReadStream(assetPath));
+
+    try {
+      return (
+        await axios.post(`${this.getBaseUrl(regionId)}/integrations/${integrationId}/assets`, formData, {
+          headers: {
+            Authorization: `Bearer ${token}`,
+            ...formData.getHeaders()
```
[the remark this answers is not in the document]
Comment (QuentinVaucherLeanIX) [reply src/client/integration.client.ts:69]: It's included in the `formData.getHeaders()` if I am not mistaken

File: src/service/leanix-region.service.ts (hunk 1 above)
Comment (QuentinVaucherLeanIX) [reply src/service/leanix-region.service.ts:6]: No issues nope. I just wouldn't expect to export anything else than a `Service` class from a `service` file, hence the `default`. But happy to remove it if that's a bad reasoning 😉

File: src/client/integration.client.ts [hunk 4]
```diff
@@ -47,12 +50,55 @@ export class IntegrationClient {
...
+  async upsertAsset(
+    regionId: string,
+    integrationId: string,
+    token: string,
+    assetPath: string,
+    contentType: string
+  ): Promise<IntegrationAssetResponseDto> {
+    const formData = new FormData();
+    formData.append('asset', fs.createReadStream(assetPath), {
+      contentType
```
Comment (QuentinVaucherLeanIX) [inline src/client/integration.client.ts:69]: @marvinrohde-leanix this should solve the content type problem we have on the UI

File: src/service/leanix-region.service.ts (hunk 1 above)
Comment (marvinrohde-leanix) [reply src/service/leanix-region.service.ts:6]: I'm fine with keeping it like that 👍🏼 
Generally, it's enough to omit `export` if you don't want to export a member.

**Pull request merged**